Sources/Utility/ArgumentParser.swift

```diff
@@ -509,27 +639,30 @@ public final class ArgumentParser {
     private func parse(_ args: [String] = [], parent: Result?) throws -> Result {
         let result = Result(parser: self, parent: parent)
         // Create options map to quickly look up the arguments.
-        let optionsTuple = options.flatMap({ option -> [(String, AnyArgument)] in
-            var result = [(option.name, option)]
-            // Add the short names too, if we have them.
-            if let shortName = option.shortName {
-                result += [(shortName, option)]
+        // FIXME: Replace with inout reduce once available
+        var optionsMap: [String:ParserOptionArgument] = [:]
+        for option in optionArguments {
+            optionsMap[option.argument.name] = option
+            if let shortName = option.argument.shortName {
+                optionsMap[shortName] = option
             }
-            return result
-        })
-        let optionsMap = Dictionary(items: optionsTuple)
```

Comment:
Initially, I wanted to edit it to use the new `Dictionary` initialisers, but the more I thought bout it, the more I went towards to procedural solution. But I can change it back, no worries.